wireapp/wire-ios-mono
# feat: reactions ordering WPB-4257

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-4257" title="WPB-4257" target="_blank"><img alt="Story" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10815?size=medium" />WPB-4257</a>  [iOS] Align order of given reaction pills
  </td></table>
  <br />
 








































----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

Reactions were sorted with the count which caused randomness and order refreshing when tapping on a message cell.
It was also causing weird behaviours as jumping reactions on message cell along with being unintuitive for the user.

### Solutions

Changed sorting algorithm to use date of creation of the reaction. The date of creation is a date when first user reacted with given emoji. When all users cancel given emoji, then the date is being reset and the emoji is treated again as fresh one if that reaction is created again.
Solution works only for reactions created after the fix, for older reactions creation date is unknown and set to distant past leading to no change in behaviour.

### Testing

#### How to Test

1. Add bunch of reactions to a message.
2. Tap on a message cell
3. The emoji ordering should not change after the tap

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
